src/components/position.cairo

```diff
@@ -0,0 +1,25 @@
+use array::ArrayTrait;
+
+#[derive(Component)]
+struct Position {
+    x: u32,
+    y: u32
+}
+
+trait PositionTrait {
+    // fn is_zero(self: Position) -> bool;
+    fn is_equal(self: Position, b: Position) -> bool;
+}
+
+impl PositionImpl of PositionTrait {
+    // fn is_zero(self: Position) -> bool {
+    //     match self.x - self.y {
+    //         0 => false,
+    //         _ => true,
+    //     }
+    // }
+
+    fn is_equal(self: Position, b: Position) -> bool {
```

Comment:
A better approach would be to impl `PartialEq` so we can then do `==` and `!=` directly.

If we need `is_zero`, you can also impl `Zeroable`.

Reply:
Interesting ! will look into that !